site/src/sphinx/client-timeouts.rst

```diff
@@ -0,0 +1,26 @@
+.. _client-timeouts:
+
+Overriding client timeouts
+==========================
+Sometimes, the default timeouts used by the Armeria-generated client does not suit a particular scenario well. In these cases, you might want to override the timeout settings.
+
+Using ``ClientOptionsBuilder``
+-----------------------------------
+
+.. code-block:: java
+
+    import com.linecorp.armeria.common.util.SafeCloseable
+    import com.linecorp.armeria.client.ClientOptionsBuilder;
+    import com.linecorp.armeria.client.Clients;
+
+    // Defaults are defined in com.linecorp.armeria.common.Flags and com.linecorp.armeria.client.ClientOptions
+    int responseTimeout = 15;
+    int writeTimeout = 1;
+
+    HelloService.Iface client = Clients.newClient("gproto+http://example.com/hello",
```

Comment:
1) `tbinary+http://...`? :-)

2) How about breaking the line after `(` so that the example does not overflow? You can preview your documentation by running `./gradlew site` at `site/build/site/index.html`.

3) You can also use `ClientBuilder` which may simplify the construction of this client:

```java
HelloService.Iface client = new ClientBuilder("tbinary+http://example.com/hello")
    .defaultResponseTimeout(Duration.ofSeconds(responseTimeout))
    .defaultWriteTimeout(Duration.ofSeconds(writeTimeout))
    .build(HelloService.Iface.class);
```

Reply:
Thanks for the feedback! :+1: Going on vacation shortly (Winter holidays here in the Nordics coming up), will try to pick this up at the end of next week.

Reply:
@trustin 

> 1. `tbinary+http://...`? :-)

What's the difference, Thrift vs gRPC? Anyway, I did the change (we are using gRPC ourselves).

> 2. How about breaking the line after `(` so that the example does not overflow? You can preview your documentation by running `./gradlew site` at `site/build/site/index.html`.

Good point, and big thanks for the preview suggestion! Made things a lot easier.

> 3. You can also use `ClientBuilder` which may simplify the construction of this client:

I added a `ClientBuilder` example now also, thanks for that. To be honest, I don't fully comprehend when `ClientBuilder` would be preferable to the `Clients.newClient` factory method. I guess they are pretty much interchangeable?

Anyway, changes done - will push in a second.

Reply:
> What's the difference, Thrift vs gRPC? Anyway, I did the change (we are using gRPC ourselves).

Yes. You used Thrift interface (`*.AsyncIface`), so I thought it is more realistic to use Thrift URI.

> I guess they are pretty much interchangeable?

Yes. I thought most users will prefer the version that used `ClientBuilder`.